File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/agent_flow_session_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class AgentFlowSessionSummary(object):
    """
    Agent flow session summary.
    """

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSessionSummary.
    #: This constant has a value of "CREATING"
    LIFECYCLE_STATE_CREATING = "CREATING"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSessionSummary.
    #: This constant has a value of "ACTIVE"
    LIFECYCLE_STATE_ACTIVE = "ACTIVE"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSessionSummary.
    #: This constant has a value of "INACTIVE"
    LIFECYCLE_STATE_INACTIVE = "INACTIVE"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSessionSummary.
    #: This constant has a value of "FAILED"
    LIFECYCLE_STATE_FAILED = "FAILED"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSessionSummary.
    #: This constant has a value of "DELETED"
    LIFECYCLE_STATE_DELETED = "DELETED"

    def __init__(self, **kwargs):
        """
        Initializes a new AgentFlowSessionSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param display_name:
            The value to assign to the display_name property of this AgentFlowSessionSummary.
        :type display_name: str

        :param key:
            The value to assign to the key property of this AgentFlowSessionSummary.
        :type key: str

        :param agent_flow_key:
            The value to assign to the agent_flow_key property of this AgentFlowSessionSummary.
        :type agent_flow_key: str

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this AgentFlowSessionSummary.
            Allowed values for this property are: "CREATING", "ACTIVE", "INACTIVE", "FAILED", "DELETED"
        :type lifecycle_state: str

        :param agent_flow_compute_key:
            The value to assign to the agent_flow_compute_key property of this AgentFlowSessionSummary.
        :type agent_flow_compute_key: str

        :param endpoint_url:
            The value to assign to the endpoint_url property of this AgentFlowSessionSummary.
        :type endpoint_url: str

        :param time_created:
            The value to assign to the time_created property of this AgentFlowSessionSummary.
        :type time_created: datetime

        :param created_by:
            The value to assign to the created_by property of this AgentFlowSessionSummary.
        :type created_by: str

        :param time_started:
            The value to assign to the time_started property of this AgentFlowSessionSummary.
        :type time_started: datetime

        :param time_ended:
            The value to assign to the time_ended property of this AgentFlowSessionSummary.
        :type time_ended: datetime

        :param duration:
            The value to assign to the duration property of this AgentFlowSessionSummary.
        :type duration: float

        :param tokens:
            The value to assign to the tokens property of this AgentFlowSessionSummary.
        :type tokens: int

        """
        self.swagger_types = {
            'display_name': 'str',
            'key': 'str',
            'agent_flow_key': 'str',
            'lifecycle_state': 'str',
            'agent_flow_compute_key': 'str',
            'endpoint_url': 'str',
            'time_created': 'datetime',
            'created_by': 'str',
            'time_started': 'datetime',
            'time_ended': 'datetime',
            'duration': 'float',
            'tokens': 'int'
        }

        self.attribute_map = {
            'display_name': 'displayName',
            'key': 'key',
            'agent_flow_key': 'agentFlowKey',
            'lifecycle_state': 'lifecycleState',
            'agent_flow_compute_key': 'agentFlowComputeKey',
            'endpoint_url': 'endpointUrl',
            'time_created': 'timeCreated',
            'created_by': 'createdBy',
            'time_started': 'timeStarted',
            'time_ended': 'timeEnded',
            'duration': 'duration',
            'tokens': 'tokens'
        }

        self._display_name = None
        self._key = None
        self._agent_flow_key = None
        self._lifecycle_state = None
        self._agent_flow_compute_key = None
        self._endpoint_url = None
        self._time_created = None
        self._created_by = None
        self._time_started = None
        self._time_ended = None
        self._duration = None
        self._tokens = None

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this AgentFlowSessionSummary.
        Display name.


        :return: The display_name of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this AgentFlowSessionSummary.
        Display name.


        :param display_name: The display_name of this AgentFlowSessionSummary.
        :type: str
        """
        self._display_name = display_name

    @property
    def key(self):
        """
        **[Required]** Gets the key of this AgentFlowSessionSummary.
        Agent Flow Session identifier.


        :return: The key of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this AgentFlowSessionSummary.
        Agent Flow Session identifier.


        :param key: The key of this AgentFlowSessionSummary.
        :type: str
        """
        self._key = key

    @property
    def agent_flow_key(self):
        """
        **[Required]** Gets the agent_flow_key of this AgentFlowSessionSummary.
        The Agent Flow key for which the session is started.


        :return: The agent_flow_key of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._agent_flow_key

    @agent_flow_key.setter
    def agent_flow_key(self, agent_flow_key):
        """
        Sets the agent_flow_key of this AgentFlowSessionSummary.
        The Agent Flow key for which the session is started.


        :param agent_flow_key: The agent_flow_key of this AgentFlowSessionSummary.
        :type: str
        """
        self._agent_flow_key = agent_flow_key

    @property
    def lifecycle_state(self):
        """
        Gets the lifecycle_state of this AgentFlowSessionSummary.
        LifecycleState of an Agent Flow Session or Deployment.

        Allowed values for this property are: "CREATING", "ACTIVE", "INACTIVE", "FAILED", "DELETED"


        :return: The lifecycle_state of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._lifecycle_state

    @lifecycle_state.setter
    def lifecycle_state(self, lifecycle_state):
        """
        Sets the lifecycle_state of this AgentFlowSessionSummary.
        LifecycleState of an Agent Flow Session or Deployment.


        :param lifecycle_state: The lifecycle_state of this AgentFlowSessionSummary.
        :type: str
        """
        allowed_values = ["CREATING", "ACTIVE", "INACTIVE", "FAILED", "DELETED"]
        if not value_allowed_none_or_none_sentinel(lifecycle_state, allowed_values):
            raise ValueError(
                "Invalid value for `lifecycle_state`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._lifecycle_state = lifecycle_state

    @property
    def agent_flow_compute_key(self):
        """
        Gets the agent_flow_compute_key of this AgentFlowSessionSummary.
        The Agent Flow Compute Key where client can run or test the Agent Flow.


        :return: The agent_flow_compute_key of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._agent_flow_compute_key

    @agent_flow_compute_key.setter
    def agent_flow_compute_key(self, agent_flow_compute_key):
        """
        Sets the agent_flow_compute_key of this AgentFlowSessionSummary.
        The Agent Flow Compute Key where client can run or test the Agent Flow.


        :param agent_flow_compute_key: The agent_flow_compute_key of this AgentFlowSessionSummary.
        :type: str
        """
        self._agent_flow_compute_key = agent_flow_compute_key

    @property
    def endpoint_url(self):
        """
        Gets the endpoint_url of this AgentFlowSessionSummary.
        The endpointUrl where the client should connect to communicate with the Agent.


        :return: The endpoint_url of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._endpoint_url

    @endpoint_url.setter
    def endpoint_url(self, endpoint_url):
        """
        Sets the endpoint_url of this AgentFlowSessionSummary.
        The endpointUrl where the client should connect to communicate with the Agent.


        :param endpoint_url: The endpoint_url of this AgentFlowSessionSummary.
        :type: str
        """
        self._endpoint_url = endpoint_url

    @property
    def time_created(self):
        """
        **[Required]** Gets the time_created of this AgentFlowSessionSummary.
        The date and time the Agent flow session was created.


        :return: The time_created of this AgentFlowSessionSummary.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this AgentFlowSessionSummary.
        The date and time the Agent flow session was created.


        :param time_created: The time_created of this AgentFlowSessionSummary.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def created_by(self):
        """
        **[Required]** Gets the created_by of this AgentFlowSessionSummary.
        The OCID of the user/principal who created the Agent flow session.


        :return: The created_by of this AgentFlowSessionSummary.
        :rtype: str
        """
        return self._created_by

    @created_by.setter
    def created_by(self, created_by):
        """
        Sets the created_by of this AgentFlowSessionSummary.
        The OCID of the user/principal who created the Agent flow session.


        :param created_by: The created_by of this AgentFlowSessionSummary.
        :type: str
        """
        self._created_by = created_by

    @property
    def time_started(self):
        """
        Gets the time_started of this AgentFlowSessionSummary.
        The date and time the session was started


        :return: The time_started of this AgentFlowSessionSummary.
        :rtype: datetime
        """
        return self._time_started

    @time_started.setter
    def time_started(self, time_started):
        """
        Sets the time_started of this AgentFlowSessionSummary.
        The date and time the session was started


        :param time_started: The time_started of this AgentFlowSessionSummary.
        :type: datetime
        """
        self._time_started = time_started

    @property
    def time_ended(self):
        """
        Gets the time_ended of this AgentFlowSessionSummary.
        The date and time the session was ended


        :return: The time_ended of this AgentFlowSessionSummary.
        :rtype: datetime
        """
        return self._time_ended

    @time_ended.setter
    def time_ended(self, time_ended):
        """
        Sets the time_ended of this AgentFlowSessionSummary.
        The date and time the session was ended


        :param time_ended: The time_ended of this AgentFlowSessionSummary.
        :type: datetime
        """
        self._time_ended = time_ended

    @property
    def duration(self):
        """
        Gets the duration of this AgentFlowSessionSummary.
        Agent flow session duration


        :return: The duration of this AgentFlowSessionSummary.
        :rtype: float
        """
        return self._duration

    @duration.setter
    def duration(self, duration):
        """
        Sets the duration of this AgentFlowSessionSummary.
        Agent flow session duration


        :param duration: The duration of this AgentFlowSessionSummary.
        :type: float
        """
        self._duration = duration

    @property
    def tokens(self):
        """
        Gets the tokens of this AgentFlowSessionSummary.
        Agent flow session token usage


        :return: The tokens of this AgentFlowSessionSummary.
        :rtype: int
        """
        return self._tokens

    @tokens.setter
    def tokens(self, tokens):
        """
        Sets the tokens of this AgentFlowSessionSummary.
        Agent flow session token usage


        :param tokens: The tokens of this AgentFlowSessionSummary.
        :type: int
        """
        self._tokens = tokens

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/AgentFlowSessionSummary.java
package com.oracle.aidataplatform.dp.model;


/**
 * Agent flow session summary.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=AgentFlowSessionSummary.Builder.class)

public final class AgentFlowSessionSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"displayName", "key", "agentFlowKey", "lifecycleState", "agentFlowComputeKey", "endpointUrl", "timeCreated", "createdBy", "timeStarted", "timeEnded", "duration", "tokens"})
    public AgentFlowSessionSummary(String displayName, String key, String agentFlowKey, DeploymentLifecycleState lifecycleState, String agentFlowComputeKey, String endpointUrl, java.util.Date timeCreated, String createdBy, java.util.Date timeStarted, java.util.Date timeEnded, Float duration, Long tokens) {
        super();
        this.displayName = displayName;
        this.key = key;
        this.agentFlowKey = agentFlowKey;
        this.lifecycleState = lifecycleState;
        this.agentFlowComputeKey = agentFlowComputeKey;
        this.endpointUrl = endpointUrl;
        this.timeCreated = timeCreated;
        this.createdBy = createdBy;
        this.timeStarted = timeStarted;
        this.timeEnded = timeEnded;
        this.duration = duration;
        this.tokens = tokens;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Display name.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("displayName")
        private String displayName;

                /**
         * Display name.
         * @param displayName the value to set
         * @return this builder
         **/
        

        public Builder displayName(String displayName) {
        this.displayName = displayName;
        return this;
        }
            /**
     * Agent Flow Session identifier.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("key")
        private String key;

                /**
         * Agent Flow Session identifier.
         * @param key the value to set
         * @return this builder
         **/
        

        public Builder key(String key) {
        this.key = key;
        return this;
        }
            /**
     * The Agent Flow key for which the session is started.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("agentFlowKey")
        private String agentFlowKey;

                /**
         * The Agent Flow key for which the session is started.
         * @param agentFlowKey the value to set
         * @return this builder
         **/
        

        public Builder agentFlowKey(String agentFlowKey) {
        this.agentFlowKey = agentFlowKey;
        return this;
        }
            /**
     * LifecycleState of an Agent Flow Session or Deployment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
        private DeploymentLifecycleState lifecycleState;

                /**
         * LifecycleState of an Agent Flow Session or Deployment.
         * @param lifecycleState the value to set
         * @return this builder
         **/
        

        public Builder lifecycleState(DeploymentLifecycleState lifecycleState) {
        this.lifecycleState = lifecycleState;
        return this;
        }
            /**
     * The Agent Flow Compute Key where client can run or test the Agent Flow.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("agentFlowComputeKey")
        private String agentFlowComputeKey;

                /**
         * The Agent Flow Compute Key where client can run or test the Agent Flow.
         * @param agentFlowComputeKey the value to set
         * @return this builder
         **/
        

        public Builder agentFlowComputeKey(String agentFlowComputeKey) {
        this.agentFlowComputeKey = agentFlowComputeKey;
        return this;
        }
            /**
     * The endpointUrl where the client should connect to communicate with the Agent.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("endpointUrl")
        private String endpointUrl;

                /**
         * The endpointUrl where the client should connect to communicate with the Agent.
         * @param endpointUrl the value to set
         * @return this builder
         **/
        

        public Builder endpointUrl(String endpointUrl) {
        this.endpointUrl = endpointUrl;
        return this;
        }
            /**
     * The date and time the Agent flow session was created.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
        private java.util.Date timeCreated;

                /**
         * The date and time the Agent flow session was created.
         * @param timeCreated the value to set
         * @return this builder
         **/
        

        public Builder timeCreated(java.util.Date timeCreated) {
        this.timeCreated = timeCreated;
        return this;
        }
            /**
     * The OCID of the user/principal who created the Agent flow session.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("createdBy")
        private String createdBy;

                /**
         * The OCID of the user/principal who created the Agent flow session.
         * @param createdBy the value to set
         * @return this builder
         **/
        

        public Builder createdBy(String createdBy) {
        this.createdBy = createdBy;
        return this;
        }
            /**
     * The date and time the session was started
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("timeStarted")
        private java.util.Date timeStarted;

                /**
         * The date and time the session was started
         * @param timeStarted the value to set
         * @return this builder
         **/
        

        public Builder timeStarted(java.util.Date timeStarted) {
        this.timeStarted = timeStarted;
        return this;
        }
            /**
     * The date and time the session was ended
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("timeEnded")
        private java.util.Date timeEnded;

                /**
         * The date and time the session was ended
         * @param timeEnded the value to set
         * @return this builder
         **/
        

        public Builder timeEnded(java.util.Date timeEnded) {
        this.timeEnded = timeEnded;
        return this;
        }
            /**
     * Agent flow session duration
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("duration")
        private Float duration;

                /**
         * Agent flow session duration
         * @param duration the value to set
         * @return this builder
         **/
        

        public Builder duration(Float duration) {
        this.duration = duration;
        return this;
        }
            /**
     * Agent flow session token usage
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("tokens")
        private Long tokens;

                /**
         * Agent flow session token usage
         * @param tokens the value to set
         * @return this builder
         **/
        

        public Builder tokens(Long tokens) {
        this.tokens = tokens;
        return this;
        }


        public AgentFlowSessionSummary build() {
            AgentFlowSessionSummary model = new AgentFlowSessionSummary(this.displayName
                    , this.key
                    , this.agentFlowKey
                    , this.lifecycleState
                    , this.agentFlowComputeKey
                    , this.endpointUrl
                    , this.timeCreated
                    , this.createdBy
                    , this.timeStarted
                    , this.timeEnded
                    , this.duration
                    , this.tokens);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(AgentFlowSessionSummary model) {
                this.displayName(model.getDisplayName());
    this.key(model.getKey());
    this.agentFlowKey(model.getAgentFlowKey());
    this.lifecycleState(model.getLifecycleState());
    this.agentFlowComputeKey(model.getAgentFlowComputeKey());
    this.endpointUrl(model.getEndpointUrl());
    this.timeCreated(model.getTimeCreated());
    this.createdBy(model.getCreatedBy());
    this.timeStarted(model.getTimeStarted());
    this.timeEnded(model.getTimeEnded());
    this.duration(model.getDuration());
    this.tokens(model.getTokens());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Display name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * Display name.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * Agent Flow Session identifier.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * Agent Flow Session identifier.
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * The Agent Flow key for which the session is started.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowKey")
    private final String agentFlowKey;

        /**
     * The Agent Flow key for which the session is started.
     * @return the value
     **/
    
    public String getAgentFlowKey() {
        return agentFlowKey;
    }

    
        /**
     * LifecycleState of an Agent Flow Session or Deployment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
    private final DeploymentLifecycleState lifecycleState;

        /**
     * LifecycleState of an Agent Flow Session or Deployment.
     * @return the value
     **/
    
    public DeploymentLifecycleState getLifecycleState() {
        return lifecycleState;
    }


        /**
     * The Agent Flow Compute Key where client can run or test the Agent Flow.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowComputeKey")
    private final String agentFlowComputeKey;

        /**
     * The Agent Flow Compute Key where client can run or test the Agent Flow.
     * @return the value
     **/
    
    public String getAgentFlowComputeKey() {
        return agentFlowComputeKey;
    }


        /**
     * The endpointUrl where the client should connect to communicate with the Agent.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("endpointUrl")
    private final String endpointUrl;

        /**
     * The endpointUrl where the client should connect to communicate with the Agent.
     * @return the value
     **/
    
    public String getEndpointUrl() {
        return endpointUrl;
    }


        /**
     * The date and time the Agent flow session was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * The date and time the Agent flow session was created.
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * The OCID of the user/principal who created the Agent flow session.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("createdBy")
    private final String createdBy;

        /**
     * The OCID of the user/principal who created the Agent flow session.
     * @return the value
     **/
    
    public String getCreatedBy() {
        return createdBy;
    }


        /**
     * The date and time the session was started
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeStarted")
    private final java.util.Date timeStarted;

        /**
     * The date and time the session was started
     * @return the value
     **/
    
    public java.util.Date getTimeStarted() {
        return timeStarted;
    }


        /**
     * The date and time the session was ended
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeEnded")
    private final java.util.Date timeEnded;

        /**
     * The date and time the session was ended
     * @return the value
     **/
    
    public java.util.Date getTimeEnded() {
        return timeEnded;
    }


        /**
     * Agent flow session duration
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("duration")
    private final Float duration;

        /**
     * Agent flow session duration
     * @return the value
     **/
    
    public Float getDuration() {
        return duration;
    }


        /**
     * Agent flow session token usage
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tokens")
    private final Long tokens;

        /**
     * Agent flow session token usage
     * @return the value
     **/
    
    public Long getTokens() {
        return tokens;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("AgentFlowSessionSummary(");
        sb.append("displayName=").append(String.valueOf(this.displayName));
        sb.append(", key=").append(String.valueOf(this.key));
        sb.append(", agentFlowKey=").append(String.valueOf(this.agentFlowKey));
        sb.append(", lifecycleState=").append(String.valueOf(this.lifecycleState));
        sb.append(", agentFlowComputeKey=").append(String.valueOf(this.agentFlowComputeKey));
        sb.append(", endpointUrl=").append(String.valueOf(this.endpointUrl));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", createdBy=").append(String.valueOf(this.createdBy));
        sb.append(", timeStarted=").append(String.valueOf(this.timeStarted));
        sb.append(", timeEnded=").append(String.valueOf(this.timeEnded));
        sb.append(", duration=").append(String.valueOf(this.duration));
        sb.append(", tokens=").append(String.valueOf(this.tokens));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AgentFlowSessionSummary)) {
            return false;
        }

        AgentFlowSessionSummary other = (AgentFlowSessionSummary) o;
        return java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.agentFlowKey, other.agentFlowKey) &&
            java.util.Objects.equals(this.lifecycleState, other.lifecycleState) &&
            java.util.Objects.equals(this.agentFlowComputeKey, other.agentFlowComputeKey) &&
            java.util.Objects.equals(this.endpointUrl, other.endpointUrl) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.createdBy, other.createdBy) &&
            java.util.Objects.equals(this.timeStarted, other.timeStarted) &&
            java.util.Objects.equals(this.timeEnded, other.timeEnded) &&
            java.util.Objects.equals(this.duration, other.duration) &&
            java.util.Objects.equals(this.tokens, other.tokens);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.agentFlowKey == null ? 43 : this.agentFlowKey.hashCode());
        result = (result * PRIME) + (this.lifecycleState == null ? 43 : this.lifecycleState.hashCode());
        result = (result * PRIME) + (this.agentFlowComputeKey == null ? 43 : this.agentFlowComputeKey.hashCode());
        result = (result * PRIME) + (this.endpointUrl == null ? 43 : this.endpointUrl.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.createdBy == null ? 43 : this.createdBy.hashCode());
        result = (result * PRIME) + (this.timeStarted == null ? 43 : this.timeStarted.hashCode());
        result = (result * PRIME) + (this.timeEnded == null ? 43 : this.timeEnded.hashCode());
        result = (result * PRIME) + (this.duration == null ? 43 : this.duration.hashCode());
        result = (result * PRIME) + (this.tokens == null ? 43 : this.tokens.hashCode());
        return result;
    }


}
